js/main.js

```diff
@@ -177,9 +178,142 @@ Site.Support = {
   },
 };
 
+Site.RadioPlayer = {
+  $radioPlayer: $('#radio-player'),
+  playerIsInitialized: false,
+
+  init: function() {
+    var _this = this;
+
+    _this.isBST = _this.checkBST();
+
+    if ($('body').hasClass('home')) {
+
+      if (_this.isSchedule()) {
+        _this.goLive();
+      }
+
+      setInterval((function() {
+        if (_this.isSchedule()) {
+          _this.goLive();
+        } else {
+          _this.goOffline();
+        }
+      }), 15000);
+
+    }
+
+  },
+
+  goLive: function() {
+    var _this = this;
+
+    if (!_this.playerIsInitialized) {
+      _this.initPlayer();
+    }
+
+    _this.$radioPlayer.show();
+  },
+
+  goOffline: function() {
+    var _this = this;
+
+    if (_this.playerIsInitialized) {
+      _this.$jPlayer.jPlayer('stop');
+    }
+
+    _this.$radioPlayer.hide();
+  },
+
+  initPlayer: function() {
+    var _this = this;
+
+    _this.$jPlayer = $('#jplayer').jPlayer({
+      ready: function () {
+        $(this).jPlayer('setMedia', {
+          title: 'resonancefm',
+          mp3: 'http://54.77.136.103:8000/resonance',
+        });
+      },
+      cssSelectorAncestor: '#jp_container_1',
+      swfPath: '/js',
+      supplied: 'mp3',
+      useStateClassSkin: true,
+      autoBlur: false,
+      smoothPlayBar: true,
+      keyEnabled: true,
+    }).bind($.jPlayer.event.play, function() {
+      $('.jp-play').hide();
+      $('.jp-stop').show();
+    }).bind($.jPlayer.event.pause, function() {
+      $('.jp-stop').hide();
+      $('.jp-play').show();
+    });
+
+    _this.playerIsInitialized = true;
+  },
+
+  isSchedule: function() {
+    var _this = this;
+
+    var utc = new Date();
+    var now;
+
+    if (_this.isBST) {
+      now = new Date(utc.getUTCFullYear(), utc.getUTCMonth(), utc.getUTCDate(), (utc.getUTCHours() + 1), utc.getUTCMinutes(), utc.getUTCSeconds());
```

Comment:
i see. the code is pretty hard to read. i couldn't rewrite it cuz i don't fully understand the schedule. 

what do? rise an issue? add a comment?